engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -2838,8 +2828,27 @@ public boolean restartNetwork(final Long networkId, final Account callerAccount,
         s_logger.debug("Restarting network " + networkId + "...");
 
         final ReservationContext context = new ReservationContextImpl(null, null, callerUser, callerAccount);
+        final NetworkOffering offering = _networkOfferingDao.findByIdIncludingRemoved(network.getNetworkOfferingId());
 
         if (cleanup) {
```

Comment:
What about extracting this IF body to a method? This allows proper documentation (instead of code comments), and unit tests.